tribe-ext-event-progress-bar.php

```diff
@@ -0,0 +1,268 @@
+
+			if ( ! $this->php_version_check() ) {
+				return;
+			}
+
+			if ( ! $this->is_using_compatible_view_version() ) {
+				return;
+			}
+
+			add_action( 'tribe_template_after_include:events/list/event/description', [ $this, 'progressbar' ], 10, 3 );
+			add_action( 'tribe_template_after_include:events/day/event/description', [ $this, 'progressbar' ], 10, 3 );
+			add_action( 'wp_enqueue_scripts', [ $this, 'safely_add_scripts' ] );
+		}
+
+		/**
+		 * Compiles the data for the Progress Bar.
+		 *
+		 * @param $file
+		 * @param $name
+		 * @param $template
+		 */
+		public function progressbar( $file, $name, $template ) {
+
+			$event = tribe_get_event( get_the_ID() );
+
+			/**
+			 * Leaving here for testing purposes.
+			 *
+			 * @todo Remove before publishing
+			 */
+			/*
+			echo date('Z' );
+			echo '<br>';
+			echo strtotime('now');
+			echo '<br>';
+			echo strtotime( $event->end_date_utc );
+			echo '<br>';
+			echo strtotime('now') - strtotime( $event->end_date_utc );
+			echo '<br>';
+			echo $event->end_date_utc;*/
+
+			// Bail if event (start) is not in the past, so event hasn't started yet
+			if ( ! $event->is_past ) {
+				/* ?>
+				<div>Event is coming.</div>
+				<?php */
+
+				return;
+			}
+
+			/**
+			 * Bail if the event has passed.
+			 *
+			 * `strtotime( 'now' )` will use the server timezone, which should be UTC.
+			 */
+			if ( strtotime( 'now' ) > strtotime( $event->end_date_utc ) ) {
+				/* ?>
+				<div>Event has passed.</div>
+				<?php */
+				return;
+			}
+
+			// Creating data for JavaScript
+			$start_date = date( 'F j, Y, g:i a', strtotime( $event->start_date_utc ) );
+			$end_date   = date( 'F j, Y, g:i a', strtotime( $event->end_date_utc ) );
+			?>
+
+			<div class="progress-bar-container progress-bar-container__on tribe-common-b2">
+				<div class="progress-bar-container__live" data-start-date="<?php echo $start_date ?>" data-end-date="<?php echo $end_date ?>">
+					<?php echo esc_html_x( 'Live now', 'Label of live event before the progress bar.', 'tribe-ext-event-progress-bar' ); ?>
+				</div>
+				<div class="progress-bar-container__background">
+					<div class="progress-bar-container__progressbar"></div>
+					<div class="progress-bar-container__ball"></div>
+				</div>
+				<div class="progress-bar-container__timeleft">
+					<?php
+					printf(
+						// translators: %1$s: The remaining time with markup, %2$s: Closing </span>.
+						esc_html_x( '%1$s left.%2$s', 'The remaining time of a live event.', 'tribe-ext-event-progress-bar' ),
+						'<span class="progress-bar-container__timeleft-time"></span> <span class="progress-bar-container__timeleft-string">',
+						'</span>'
+					);
+					?>
+				</div>
+			</div>
+
+			<?php
+		}
+
+		/**
+		 * Check if we have a sufficient version of PHP. Admin notice if we don't and user should see it.
+		 *
+		 * @return bool
+		 */
+		private function php_version_check() {
+			$php_required_version = '7.0';
```

Comment:
Add comment why the version you required.

Reply:
Bad habit of mine. :)
Done.